### PR TITLE
Add option to make active list filters more visible

### LINKED ISSUE #174: Make currently active list filters more visible
On a list page with many filters shown and the dropdown option active, admin users have a hard time finding out which filters are currently active, in order to remove one for example if the current selection has no results.  They would have to go through the list and determine if the value visible is the default or not, so they end up clearing all filters (it’s the only action displayed) and re-building their selection.

Ideas:
- A: add CSS to highlight filters with a non-default value selected

- B: put active filters at the top of the list of filters

- C: on top of the filter list, show currently active filters with a link to remove each: `( Age: 9+  | x ) ( Group: 10-15  | x )`  (parens to represent a chip widget)

A + B or A + C are possible combinations.

I can contribute code if you agree to this feature!

### PR DESCRIPTION
**Pull request by merwok**
Fixes #174

### CODE REVIEW

**Comment by merwok** [comment]
Reused module colours for default style, bold only for dropdowns:

dai-filters-default  dai-filters-dropdown

File: admin_interface/templates/admin/filter.html [hunk 1]
```diff
@@ -10,13 +10,12 @@
 
     {# Use the default list filter template -> https://github.com/django/django/blob/master/django/contrib/admin/templates/admin/filter.html #}
 
-    <h3>{% blocktrans with filter_title=title %} By {{ filter_title }} {% endblocktrans %}</h3>
-    <ul>
+    <h3 {% if spec.lookup_val %}class="active"{% endif %}>{% blocktrans with filter_title=title %} By {{ filter_title }} {% endblocktrans %}</h3>
```
Comment (merwok) [inline admin_interface/templates/admin/filter.html:13]: `spec` is the list filter instance, `lookup_val` is the value in the query string

Comment (fabiocaccamo) [reply admin_interface/templates/admin/filter.html:13]: Thanks for having explained it in advance.

**Review by fabiocaccamo — APPROVED** [review]

File: admin_interface/templates/admin/filter.html (hunk 1 above)
Comment (merwok) [reply admin_interface/templates/admin/filter.html:13]: Should it be a comment in the code?

**Comment by merwok** [comment]
The PR contains item A from the discussion, I will add item C too!

File: admin_interface/templates/admin/filter.html (hunk 1 above)
Comment (fabiocaccamo) [reply admin_interface/templates/admin/filter.html:13]: I think the purpose can be easily guessed by reading the code, however a comment never hurts.

Comment (merwok) [reply admin_interface/templates/admin/filter.html:13]: Guessing from context is good enough, and I don’t want to have to think about whether a comment should be in which file or both so none 🙂

File: admin_interface/templates/admin/filter.html [hunk 2]
```diff
@@ -8,15 +8,14 @@
 
 {% else %}
 
-    {# Use the default list filter template -> https://github.com/django/django/blob/master/django/contrib/admin/templates/admin/filter.html #}
+    {# Use a changed default list filter template -> https://github.com/django/django/blob/master/django/contrib/admin/templates/admin/filter.html #}
 
-    <h3>{% blocktrans with filter_title=title %} By {{ filter_title }} {% endblocktrans %}</h3>
-    <ul>
+    <h3 {% if spec.lookup_val or spec.value %}class="active"{% endif %}>{% blocktrans with filter_title=title %} By {{ filter_title }} {% endblocktrans %}</h3>
+    <ul {% if spec.lookup_val or spec.value %}class="active"{% endif %}>
```
Comment (merwok) [inline admin_interface/templates/admin/filter.html:14]: Some custom filters don’t have `lookup_val`, so this calls the normal `value` method.
`value` alone is not enough so I kept both; performance impact should be minimal.

File: admin_interface/templates/django-admin-autocomplete-filter/autocomplete-filter.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{#
+ copied from https://github.com/farhan0581/django-admin-autocomplete-filter/ to highlight active
```
Comment (merwok) [inline admin_interface/templates/django-admin-autocomplete-filter/autocomplete-filter.html:2]: Can’t copy this, the project is under GPL!

Comment (fabiocaccamo) [reply admin_interface/templates/django-admin-autocomplete-filter/autocomplete-filter.html:2]: You can rewrite the same logic without copying it directly.

**Comment by fabiocaccamo** [comment]
> I also wanted feedback about the styles from you and from my client who’s funding this work 

My personal feedback on style:

- I think the padding between the selected option and the list filter container should be the same on left and right.

- If there is horizontal space between the selected option and the container, consider rounding border radius of the selected option using the css variable.

- On the left of the selected item, a very soft grey marker is still visible, when list filter highlight is enabled it should be hidden or of the same color of the selected option.

---

I'm glad for you that your client is funding this work! 
As you can see this project is widely used, but unfortunately it has few contributors and no sponsors at all.

**Comment by fabiocaccamo** [comment]
> The PR contains item A from the discussion, I will add item C too!

I think now that item A and C are two different features and probably it's better to have the possibility to enable them separately, do you agree?

**Comment by merwok** [comment]
> it's better to have the possibility to enable them separately

I was already planning to have two different settings, and we can also have two PRs.

File: admin_interface/templates/django-admin-autocomplete-filter/autocomplete-filter.html (hunk 3 above)
Comment (merwok) [reply admin_interface/templates/django-admin-autocomplete-filter/autocomplete-filter.html]: To override the template I need to copy and change it, so I won’t be taking any risk for your project and handle the override in my site.

**Comment by merwok** [comment]
Thanks for the feedback!  I will try to apply it after a weekend, unless you want to do it quickly and add commits to this branch.

**Comment by fabiocaccamo** [comment]
No hurry, have a nice weekend!

**Comment by fabiocaccamo** [comment]
@merwok maybe we could start to merge this PR?

**Comment by merwok** [comment]
I think the CSS tweaks need to be addressed!

**Review by fabiocaccamo — DISMISSED** [review]

**Comment by fabiocaccamo** [comment]
@merwok I agree, I wait for the CSS changes.

**Comment by merwok** [comment]
I have applied changes for the three points:

list-filters-revised


For this one:
> If there is horizontal space between the selected option and the container,

I’m not sure what you mean by that, so I just added a border radius unconditionally (if the highlight option is enabled).

**Comment by fabiocaccamo** [comment]
I looks perfect to me!
I noticed only now that I reviewed the PR while it was still in draft, sorry.

---

> I’m not sure what you mean by that, so I just added a border radius unconditionally (if the highlight option is enabled).

I meant this padding: 

194202262-6daf02ad-5d68-4825-9c13-3aebe71fce3d

**Pull request merged**